### ISSUE TITLE
FR: Basic UI: Adjust webview height automatically

### ISSUE THREAD

**Issue by mueller-ma** [eclipse/smarthome#6077]
If no height attribute for a webview item is set, a very small frame is shown in Basic UI:
````
Webview url="https://www.wikipedia.org/"
````
![grafik](https://user-images.githubusercontent.com/22525368/44306756-684afe80-a395-11e8-88a4-72992d45d5e0.png)

It can be difficult to set the height, so the best solution would be IMO to set no height at all and let the UI maximize the webview. This is already implement - by accident - in the openHAB app for Android.

**Comment by kaikreuzer** [comment]
@resetnow Something for you?

**Comment by resetnow** [comment]
Unfortunately, this can't be implemented easily due to how browsers work. In OpenHAB, webviews are rendered as `iframe`. Cross-origin safety rules implemented in browsers prohibit any access from the "top" document to "child" document, including to such parameters as height and width. The only working solution involves willing interaction from the "child", which could send a message with its height to the parent. This requires inserting scripts into the page, dealing with HTTPS and so on.

>This is already implement - by accident - in the openHAB app for Android.

This is correct — but in Android, the app has more access to webview inside it. In BasicUI, UI page itself and webview inside it are on more or less equal terms and are separated by browser safety policies.

**Comment by mueller-ma** [comment]
I see. Is it possible to increase the default height? This way it's possible to have a working sitemap for both clients (Basic ui and Android app).
Otherwise you can either have a fixed height and have a hardly usable webview in Android (because you cannot scroll and from a UX perspective it's hard to implement scolling) or you can omit the height attribute and have a unusable webview in basic ui.

**Comment by resetnow** [comment]
Ideally, it would be good to have support for auto height with fallback to numeric value; for example `width="auto,4"`. Clients could then do their best to get actual height and if that's not possible, use fallback. I don't know how intuitive it is though.

**Comment by mueller-ma** [comment]
Is there any actuall usecase where a user wants to set a height and not see the complete website? Wouldn't it be better when Android app ignores the height attribute and always show the full page. And UIs that cannot auto-scale use the height?

**Comment by resetnow** [comment]
@kaikreuzer what do you think about `height="auto,4"` ?

**Comment by kaikreuzer** [comment]
Could we treat the "height" parameter as a "maxHeight"? I.e. always do auto-scale, but allow the user to limit the height, because the content might e.g. be a long news feed.
This logic should be valid for Basic UI and native apps likewise, wdyt?

**Comment by resetnow** [comment]
This perhaps depends on whether or not it's possible to do in Android's WebView in a "clean" way — without layout changing as page continues to load, which may be annoying.

**Comment by kaikreuzer** [comment]
Let's see what @mueller-ma says about that. But if I got him right, he wants to have auto-height, so that should just be fine.

**Comment by mueller-ma** [comment]
I've done some research, but haven't found a clean way to limit the height of a webview. Maybe @maniac103 knows a smart way.

**Comment by maniac103** [comment]
Max height should be totally doable in Android by creating a custom WebView subclass overriding onMeasure(), like seen [here](https://stackoverflow.com/questions/18425758/how-to-set-a-maximum-height-with-wrap-content-in-android)

**Comment by mueller-ma** [comment]
> Could we treat the "height" parameter as a "maxHeight"?

Should I start implementing this in the app?

**Comment by kaikreuzer** [comment]
I'm all for it 👍

**Comment by mueller-ma** [comment]
I opened a PR. I guess the docs should be changed as well?

**Comment by mueller-ma** [comment]
https://github.com/openhab/openhab-android/releases/tag/2.3.6-beta contains this change.

**Comment by mueller-ma** [comment]
I guess the default hight of the webview in basic ui should be changed. The current default is too low.

**Comment by kaikreuzer** [comment]
I'd agree.

**Comment by kaikreuzer** [comment]
@resetnow Wdyt?

**Comment by resetnow** [comment]
Yes, I guess default height can be changed so the widget has approximately 16:9 or 4:3 ratio
